assets/src/phoenix/session_timeout.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { Modal } from 'react-bootstrap';
+import ReactDOM from 'react-dom';
+
+// Constant value used to configure the time before the warning message is displayed
+// ms * sec * min example: 1000 * 60 * 5 = 5 minutes
+const timeBeforeWarningMs = 1000 * 60 * 5;
+
+// This is the entry point for the session timeout feature.
+export function sessionTimeout() {
+  setTimeout(triggerTimers, 1000);
+}
```

Comment:
This is the entry function that triggers the timers. Note that iInstead of directly calling `triggerTimers`, I used a timeout because I discovered some race conditions when reading the cookie.